common/src/main/kotlin/entity/Interactions.kt

```diff
@@ -0,0 +1,239 @@
+}
+
+
+@Serializable(ApplicationCommandOptionType.Serializer::class)
+sealed class ApplicationCommandOptionType(val type: Int) {
+
+
+    object SubCommand : ApplicationCommandOptionType(1)
+    object SubCommandGroup : ApplicationCommandOptionType(2)
+    object String : ApplicationCommandOptionType(3)
+    object Integer : ApplicationCommandOptionType(4)
+    object Boolean : ApplicationCommandOptionType(5)
+    object User : ApplicationCommandOptionType(6)
+    object Channel : ApplicationCommandOptionType(7)
+    object Role : ApplicationCommandOptionType(8)
+    object Unknown : ApplicationCommandOptionType(Int.MAX_VALUE)
+
+    companion object
+
+    object Serializer : KSerializer<ApplicationCommandOptionType> {
+
+        override val descriptor: SerialDescriptor
+            get() = PrimitiveSerialDescriptor("ApplicationCommandOptionType", PrimitiveKind.INT)
+
+        override fun deserialize(decoder: Decoder): ApplicationCommandOptionType {
+            return when (decoder.decodeInt()) {
+                1 -> SubCommand
+                2 -> SubCommandGroup
+                3 -> String
+                4 -> Integer
+                5 -> Boolean
+                6 -> User
+                7 -> Channel
+                8 -> Role
+                else -> Unknown
+            }
+        }
+
+        override fun serialize(encoder: Encoder, value: ApplicationCommandOptionType) {
+            encoder.encodeInt(value.type)
+        }
+    }
+
+
+}
+
+@Serializable(Choice.ChoiceSerializer::class)
+sealed class Choice<out T> {
+    abstract val name: String
+    abstract val value: T
+
+    class IntChoice(override val name: String, override val value: Int) : Choice<Int>()
+    class StringChoice(override val name: String, override val value: String) : Choice<String>()
+    internal class ChoiceSerializer<T>(serializer: KSerializer<T>) : KSerializer<Choice<*>> {
+        override val descriptor: SerialDescriptor = buildClassSerialDescriptor("Choice") {
+            element<String>("name")
+            element<String>("value")
+        }
+
+        override fun deserialize(decoder: Decoder): Choice<*> {
+            lateinit var name: String
+            lateinit var value: JsonPrimitive
+            val json = decoder as JsonDecoder
+            with(decoder.beginStructure(descriptor) as JsonDecoder) {
+                while (true) {
+                    when (val index = decodeElementIndex(descriptor)) {
+                        0 -> name = decodeStringElement(descriptor, index)
+                        1 -> value = decodeJsonElement().jsonPrimitive
+
+                        CompositeDecoder.DECODE_DONE -> break
+                        else -> throw SerializationException("unknown index: $index")
+                    }
+                }
+                endStructure(descriptor)
+            }
+            return if(value.isString) StringChoice(name, value.toString()) else IntChoice(name, value.int)
```

Comment:
I think it'd be better if we gave a proper error when the type isn't string nor int. This shouldn't happen (yet?), but it'll probably save us some work when they eventually break this stuff.